src/routing/kad-dht.md

```diff
@@ -0,0 +1,719 @@
+---
+title: IPFS Kademlia DHT
+description: >
+  The IPFS Distributed Hash Table (DHT) specification defines a structured
+  overlay network used for peer and content routing in the InterPlanetary File
+  System (IPFS). It extends the libp2p Kademlia DHT specification, adapting and
+  adding features to support IPFS-specific requirements.
+date: 2025-03-24
+maturity: reliable
+editors:
+  - name: Guillaume Michel
+    url: https://guillaume.michel.id
+    github: guillaumemichel
+    affiliation:
+      name: Shipyard
+      url: https://ipshipyard.com
+tags: ['routing']
+order: 1
+---
+
+The IPFS Distributed Hash Table (DHT) specification defines a structured
+overlay network used for peer and content routing in the InterPlanetary File
+System (IPFS). It extends the libp2p Kademlia DHT specification, adapting and
+adding features to support IPFS-specific requirements.
+
+## Introduction
+
+The Kademlia Distributed Hash Table (DHT) is a decentralized key-value store
+designed to enable efficient and scalable peer-to-peer routing. It provides a
+structured overlay network that allows nodes to locate peers and content in a
+distributed system without relying on centralized servers.
+
+The primary goal of the Kademlia routing algorithm is to progressively discover
+and interact with nodes that are closest to a given key based on the network's
+distance metric. Once a node has identified the closest peers, it can either:
+
+* **Locate a specific peer** in the network
+* **Find content providers** serving content associated with a CID
+* **Store and retrieve values** directly within the DHT, such as IPNS names
+
+### Relation to libp2p kad-dht
+
+The IPFS Kademlia DHT specification is a specialization of the [libp2p Kademlia DHT](https://github.com/libp2p/specs/tree/master/kad-dht).
```

Comment:
It seems like an awful lot of this document isn't describing anything specialized or specific to IPFS or the Amino DHT and is general to the libp2p kad spec.

What's the approach we're trying to take here:
1. Only add IPFS / Amino -specific info
2. Copy-paste an independent document so IPFS / Amino users don't need to look at the libp2p kad spec
3. Make a mostly independent IPFS / Amino doc so users only have to look at the libp2p one for levels of detail exceeding some threshold
4. Something else

This paragraph makes it sound like 1 is the approach we're taking but the doc looks like 2 or 3.

Reply:
I'm ok with 2-3, for reasons listed in https://github.com/ipfs/specs/pull/497#discussion_r2019187655

Perhaps go further and change this paragraph to state "The IPFS Kademlia DHT specification extends [libp2p Kademlia DHT](https://github.com/libp2p/specs/tree/master/kad-dht) with features related to CID, IPNS, and content providing."

Reply:
https://github.com/ipfs/specs/pull/497/commits/27d2a5825816422c50ca1f462ddb4355784fc982